packages/itwinui-css/src/input-container/input-container.scss

```diff
@@ -192,6 +188,20 @@
     }
   }
 
+  // Collapse padding between input/textarea and .iui-input-flex-container-icon (Icon or IconButton)
+  > :is(input, textarea):has(+ .iui-input-flex-container-icon) {
+    padding-inline-end: 0;
+  }
+
+  > :is(.iui-input-flex-container-icon) + :is(input, textarea) {
+    padding-inline-start: 0;
+  }
+
+  // input/textarea is not the first child and does not directly follow .iui-input-flex-container-icon (Icon or IconButton)
+  > :not(.iui-input-flex-container-icon) + :is(input, textarea) {
+    padding-inline-start: var(--iui-size-2xs);
```

Comment:
What specific case is this handling? Isn't it the general case, where we should set `padding-inline` on all `:is(input,textarea)`?

Reply:
This `padding-inline-start` was initially in the general case (`input, textarea`). I moved it in a separate selector to prevent conflicts between the two selectors that set `padding-inline-start`.

The specificity difference between the [general selector](https://github.com/iTwin/iTwinUI/pull/1881/files#diff-79a618b1da06b9cedff0df3a138d8a8da22a8185e8f0b5a2cf6c8aeb23a49b5cR167-R168) and the [specialized selector](https://github.com/iTwin/iTwinUI/pull/1881/files#diff-79a618b1da06b9cedff0df3a138d8a8da22a8185e8f0b5a2cf6c8aeb23a49b5cR196) resolves the conflict correctly, but I felt being more explicit in the selector might be safer here.

Reply:
> but I felt being more explicit in the selector might be safer here.

It's not safer, it introduces extra unnecessary complexity. We should utilize the power of the CSS cascade where we can. The most specific rules should be at the bottom.

Here's how I think of it, from top to bottom:

1. Most general case where input is used by itself (same as `main` and this PR):

https://github.com/iTwin/iTwinUI/blob/00cef97bad6e912c74f22328826d85e384c7149c/packages/itwinui-css/src/input-container/input-container.scss#L177

2. Less specific case where input has things before or after it:

(same as `main` branch, missing in this PR)

https://github.com/iTwin/iTwinUI/blob/00cef97bad6e912c74f22328826d85e384c7149c/packages/itwinui-css/src/input-container/input-container.scss#L180-L182

(missing in `main` and this PR)

```css
&:where(:not(:last-child)) {
  padding-inline-end: var(--iui-size-2xs);
}
```

3. Most specific case (this PR)

https://github.com/iTwin/iTwinUI/blob/1d5897fff1208b0706d36ef9e6a1ddd5b5068465/packages/itwinui-css/src/input-container/input-container.scss#L191-L198

Reply:
I had changed to the more specific selector since I felt relying on specificities to resolve selector conflicts is brittle and could lead to unexpected issues down the line.

But I agree with preferring simplicity part. I don't feel strongly in any direction, so I moved the specific selector into the general case (https://github.com/iTwin/iTwinUI/pull/1881/commits/860771ce726674f54b521bcebbbd2ac1b29090f1). I also added the last child selector that was missing from before this PR.